### PR TITLE
[MAINTENANCE] Add support for automatically building expectations gallery metadata

### PR DESCRIPTION
**Pull request by jcampbell**

### CODE REVIEW

File: assets/scripts/build_gallery.py [hunk 1]
```diff
@@ -1,11 +1,95 @@
...
+chandler = logging.StreamHandler(stream=sys.stdout)
+chandler.setLevel(logging.DEBUG)
+chandler.setFormatter(
+    logging.Formatter("%(asctime)s - %(levelname)s - %(message)s", "%Y-%m-%dT%H:%M:%S")
+)
+logger.addHandler(chandler)
+logger.setLevel(logging.DEBUG)
+
+
+def execute_shell_command(command: str) -> int:
```
Comment (spbail) [inline assets/scripts/build_gallery.py:23]: How is this different from subprocess.run()? (If there's some relevant different, I'd add an explanation to the docstring)

File: assets/scripts/build_gallery.py [hunk 2]
```diff
@@ -1,11 +1,95 @@
...
+        exception_traceback: str = traceback.format_exc()
+        exception_message += (
+            f'{type(cpe).__name__}: "{str(cpe)}".  Traceback: "{exception_traceback}".'
+        )
+        logger.error(exception_message)
+
+    return status_code
+
+
+def get_contrib_requirements(filepath):
```
Comment (spbail) [inline assets/scripts/build_gallery.py]: Add docstring pls?

File: assets/scripts/build_gallery.py [hunk 3]
```diff
@@ -30,7 +114,21 @@ def build_gallery(include_core=True, include_contrib_experimental=True):
...
+        for root, dirs, files in os.walk(contrib_experimental_dir):
+            for file in files:
+                if file.endswith(".py") and not file == "__init__.py":
+                    logger.debug(f"Getting requirements for module {file}")
+                    requirements_dict[file[:-3]] = get_contrib_requirements(
+                        os.path.join(root, file)
+                    )
+
         for expectation_module in expectations_module.__all__:
+            if expectation_module in requirements_dict:
```
Comment (spbail) [inline assets/scripts/build_gallery.py:179]: Nitpick, the install step for expectation and metric dependencies could be moved into a separate method.

File: assets/scripts/build_gallery.py [hunk 4]
```diff
@@ -1,11 +1,95 @@
...
+
+def get_contrib_requirements(filepath):
+    with open(filepath) as file:
+        tree = ast.parse(file.read())
+
+    requirements_info = {"classes": [], "requirements": []}
+    for child in ast.iter_child_nodes(tree):
+        if not isinstance(child, ast.ClassDef):
+            continue
+        requirements_info["classes"] += child.name
```
Comment (spbail) [inline assets/scripts/build_gallery.py]: I don't see this being used anywhere?

**Review by spbail — COMMENTED** [review]
I have questions!

File: assets/scripts/build_gallery.py (hunk 1 above)
Comment (jcampbell) [reply assets/scripts/build_gallery.py:23]: Added a note in the docstring -- it simply provides Exception handling

File: assets/scripts/build_gallery.py (hunk 2 above)
Comment (jcampbell) [reply assets/scripts/build_gallery.py]: Done!

File: assets/scripts/build_gallery.py (hunk 3 above)
Comment (jcampbell) [reply assets/scripts/build_gallery.py:179]: I realized that we don't have run_diagnostics for metrics, so clarified that and added a warning if one is discovered.

File: assets/scripts/build_gallery.py (hunk 4 above)
Comment (jcampbell) [reply assets/scripts/build_gallery.py]: Added a note on the intended future use (and improved its organization). Good call.

**Review by spbail — APPROVED** [review]
lgtm, thanks for making the changes

**Pull request merged**